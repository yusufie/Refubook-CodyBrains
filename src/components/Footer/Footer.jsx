import React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import logo from '../../assets/logo.svg';
import { LanguagesMenu } from '..';

const Footer = () => {
  const { t } = useTranslation();
  return (
    <div className="flex flex-col  gap-5 px-14 py-6 mb-6 lg:flex-row  w-full bg-refubookWhite mt-10">
      <div className="items-center justify-center flex mr-6">
        <Link
          to="/"
          className="text-refubookBlue font-bold text-xl tracking-tight "
        >
          <img src={logo} alt="logo" className="w-10 h-10" />
        </Link>
      </div>

      <ul className="flex justify-center gap-5 items-center text-md md:text-lg xl:text-xl 2xl:text-2xl  w-full">
        <li>
          <Link
            to="/"
            className="text-refubookGray hover:text-refubookActiveNav font-medium"
          >
            {t('Home')}
          </Link>
        </li>

        <li>
          <Link
            to="/about"
            className="text-refubookGray hover:text-refubookActiveNav font-medium"
          >
            {t('About')}
          </Link>
        </li>

        <li>
          <Link
            to="/contact"
            className="text-refubookGray hover:text-refubookActiveNav font-medium"
          >
            {t('Contact')}
          </Link>
        </li>

        <li>
          <Link
            to="/blog"
            className="text-refubookGray hover:text-refubookActiveNav font-medium"
          >
            {t('Blog')}
          </Link>
        </li>
      </ul>

      <div className="flex w-full justify-center gap-5 md:gap-10  lg:justify-end">
        <Link to="/signup">
          <button
            type="button"
            className=" flex items-center w-32 md:w-full justify-center font-medium z-40 text-xs md:text-lg bg-refubookBlue text-refubookWhite rounded-full py-3 md:py-2.5  md:px-10 px-6 cursor-pointer  duration-300 ease-in"
          >
            {t('Sign_Up')}
          </button>
        </Link>
        <LanguagesMenu />
      </div>
    </div>
  );
};

export default Footer;
