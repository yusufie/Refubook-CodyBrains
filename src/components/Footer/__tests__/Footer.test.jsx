import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

jest.mock('react-router-dom', () => ({
  Link: 'Link',
}));
describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
