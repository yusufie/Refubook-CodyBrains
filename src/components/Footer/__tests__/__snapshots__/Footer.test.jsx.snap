// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="flex flex-col  gap-5 px-14 py-6 mb-6 lg:flex-row  w-full bg-refubookWhite mt-10"
>
  <div
    className="items-center justify-center flex mr-6"
  >
    <Link
      className="text-refubookBlue font-bold text-xl tracking-tight "
      to="/"
    >
      <img
        alt="logo"
        className="w-10 h-10"
        src="logo.svg"
      />
    </Link>
  </div>
  <ul
    className="flex justify-center gap-5 items-center text-md md:text-lg xl:text-xl 2xl:text-2xl  w-full"
  >
    <li>
      <Link
        className="text-refubookGray hover:text-refubookActiveNav font-medium"
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        className="text-refubookGray hover:text-refubookActiveNav font-medium"
        to="/about"
      >
        About
      </Link>
    </li>
    <li>
      <Link
        className="text-refubookGray hover:text-refubookActiveNav font-medium"
        to="/contact"
      >
        Contact
      </Link>
    </li>
    <li>
      <Link
        className="text-refubookGray hover:text-refubookActiveNav font-medium"
        to="/blog"
      >
        Blog
      </Link>
    </li>
  </ul>
  <div
    className="flex w-full justify-center gap-5 md:gap-10  lg:justify-end"
  >
    <Link
      to="/signup"
    >
      <button
        className=" flex items-center w-32 md:w-full justify-center font-medium z-40 text-xs md:text-lg bg-refubookBlue text-refubookWhite rounded-full py-3 md:py-2.5  md:px-10 px-6 cursor-pointer  duration-300 ease-in"
        type="button"
      >
        Sign_Up
      </button>
    </Link>
    <div
      className="flex items-center justify-center relative"
    >
      <button
        aria-expanded={false}
        aria-haspopup="listbox"
        className="  flex items-center justify-between gap-2 font-medium z-40  bg-refubookBlue text-refubookWhite rounded-full py-2  md:px-10 px-3 cursor-pointer  duration-300 ease-in"
        data-headlessui-state=""
        disabled={false}
        id="headlessui-listbox-button-:r0:"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="text-sm md:text-lg "
        >
           
          English
        </span>
        <span
          className="flag-icon flag-icon-gb mx-2"
        />
        <svg
          className="w-5 h-5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.646 4.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1-.708.708L8 5.707l-5.646 5.647a.5.5 0 0 1-.708-.708l6-6z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
